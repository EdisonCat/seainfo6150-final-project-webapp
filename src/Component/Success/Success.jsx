import React from "react";
import styles from "../SignUp/SignUp.module.css";
import { useHistory } from "react-router-dom";
export default function Success() {
	const history = useHistory();
	const handleClick = () => {
		history.push({ pathname: "/" });
	};
	return (
		<div className={styles.container}>
			<div>
				<h1>
					Thanks for signing up,{" "}
					{history.location.state
						? history.location.state.username
						: "user"}
					!
					<br />
					{history.location.state ? (
						history.location.state.join.length > 0 ? (
							<>
								You've joined our{" "}
								<i>{history.location.state.join}</i> event,
								please check your inbox for more information!
								<br />
							</>
						) : null
					) : null}
					You are now one of our{" "}
					{history.location.state
						? history.location.state.type
						: ""}{" "}
					account members!
					<br />
					You've subscribed to our notification emails for upcoming
					{history.location.state
						? history.location.state.event
							? " events"
							: null
						: null}
					{history.location.state
						? history.location.state.movies
							? " movies"
							: null
						: null}
					{history.location.state
						? history.location.state.tv
							? " TV shows"
							: null
						: null}
					{history.location.state
						? history.location.state.comics
							? " comics"
							: null
						: null}
				</h1>
				<button onClick={() => handleClick()}>Go to home page</button>
			</div>
		</div>
	);
}