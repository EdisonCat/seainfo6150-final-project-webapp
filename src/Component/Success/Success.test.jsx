import React from "react";
import Success from "./Success";
it("rendered Success correctly", () => {
	// eslint-disable-next-line no-undef
    const { container } = render(<Success />);
    expect(container).toMatchSnapshot();
});