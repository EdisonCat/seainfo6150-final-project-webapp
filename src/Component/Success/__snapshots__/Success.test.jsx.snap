// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered Success correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      <h1>
        Thanks for signing up,
         
        user
        !
        <br />
        You are now one of our
         
        
         
        account members!
        <br />
        You've subscribed to our notification emails for upcoming
      </h1>
      <button>
        Go to home page
      </button>
    </div>
  </div>
</div>
`;
